homeassistant/helpers/entity_platform.py

```diff
@@ -52,22 +51,21 @@ def __init__(self, *, hass, logger, domain, platform_name, platform,
         # which powers entity_component.add_entities
         if platform is None:
             self.parallel_updates = None
+            self.parallel_updates_semaphore = None
             return
 
-        # Async platforms do all updates in parallel by default
-        if hasattr(platform, 'async_setup_platform'):
-            default_parallel_updates = 0
-        else:
-            default_parallel_updates = 1
-
-        parallel_updates = getattr(platform, 'PARALLEL_UPDATES',
-                                   default_parallel_updates)
+        self.parallel_updates = getattr(platform, 'PARALLEL_UPDATES', None)
+        # semaphore will be created on demand
+        self.parallel_updates_semaphore = None
 
-        if parallel_updates:
-            self.parallel_updates = asyncio.Semaphore(
-                parallel_updates, loop=hass.loop)
-        else:
-            self.parallel_updates = None
+    def _get_parallel_updates_semaphore(self):
+        """Get or create a semaphore for parallel updates."""
+        if self.parallel_updates_semaphore is None:
```

Comment:
Why not make this 
```suggestion
        if self.parallel_updates_semaphore is None and self.parallel_updates:
```

Reply:
Because for sync platform, `paralleLupdates` is None means we need a semaphore. See my decision table in the later code comment